lib/async_hooks.js

```diff
@@ -339,10 +322,14 @@ function emitInitS(asyncId, type, triggerId, resource) {
 
 function emitBeforeN(asyncId) {
   processing_hook = true;
-  for (var i = 0; i < active_hooks_array.length; i++) {
-    if (typeof active_hooks_array[i][before_symbol] === 'function') {
-      runCallback(active_hooks_array[i][before_symbol], asyncId);
+  try {
+    for (var i = 0; i < active_hooks_array.length; i++) {
```

Comment:
@refack That's no longer an issue, which is why the index is no longer used. At the top of the file you'll see:
```js
// Use to temporarily store and updated active_hooks_array if the user enables
// or disables a hook while hooks are being processed.
var tmp_active_hooks_array = null;
// Keep track of the field counts held in tmp_active_hooks_array.
var tmp_async_hook_fields = null;
```
Which `getHooksArray()` automatically assigns the old array to and duplicates the new altered array. So it's impossible for the user to alter the hooks mid-execution. i.e. you'll never see the case of "`a.push()`" to the actual array that contains the hooks in the middle of execution. So feel free to benchmark it all you want.